client/rpc/txpool/src/lib.rs

```diff
@@ -0,0 +1,154 @@
+// Copyright 2019-2020 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+use ethereum::{Transaction as EthereumTransaction, TransactionMessage};
+use ethereum_types::{H160, H256, U256};
+use jsonrpc_core::Result as RpcResult;
+use jsonrpc_core::{Error, ErrorCode};
+pub use moonbeam_rpc_core_txpool::{
+	GetT, Summary, Transaction, TransactionMap, TxPool as TxPoolT, TxPoolResult, TxPoolServer,
+};
+use sha3::{Digest, Keccak256};
+use sp_api::{BlockId, ProvideRuntimeApi};
+use sp_blockchain::{Error as BlockChainError, HeaderBackend, HeaderMetadata};
+use sp_runtime::traits::Block as BlockT;
+use sp_transaction_pool::{InPoolTransaction, TransactionPool};
+use std::collections::HashMap;
+use std::{marker::PhantomData, sync::Arc};
+
+use moonbeam_rpc_primitives_txpool::TxPoolRuntimeApi;
+
+// Note: both `internal_err` and `public_key` could be imported from Frontier.
+// However there are dependency collisions due to Frontier using a different Substrate branch.
+// This may change in the future, and in that case we could just import them.
+pub fn internal_err<T: ToString>(message: T) -> Error {
+	Error {
+		code: ErrorCode::InternalError,
+		message: message.to_string(),
+		data: None,
+	}
+}
+
+pub fn public_key(
+	transaction: &EthereumTransaction,
+	hash: H256,
+) -> Result<[u8; 64], sp_io::EcdsaVerifyError> {
+	let mut sig = [0u8; 65];
+	let mut msg = [0u8; 32];
+	sig[0..32].copy_from_slice(&transaction.signature.r()[..]);
+	sig[32..64].copy_from_slice(&transaction.signature.s()[..]);
+	sig[64] = transaction.signature.standard_v();
+	msg.copy_from_slice(&hash[..]);
+
+	sp_io::crypto::secp256k1_ecdsa_recover(&sig, &msg)
+}
+
+pub struct TxPool<B: BlockT, C, P> {
```

Comment:
Same, a comment would help a lot. Why and how is this different than the substrate transaction pool that it encapsulates?